core/src/main/java/com/linecorp/armeria/server/AggregatingDecodedHttpRequest.java

```diff
@@ -147,11 +147,17 @@ public boolean tryWrite(HttpObject obj) {
         if (obj instanceof HttpData) {
             ((HttpData) obj).touch(routingCtx);
             if (obj.isEndOfStream()) {
+                if (ctx != null) {
+                    ctx.logBuilder().requestFullyReceived();
+                }
```

Comment:
There are two problems if we rely on `isEndOfStream()` to log `requestFullyReceived`
- The last data for HTTP/1 does not have `endOfStream` flag.
- Not all requests finish successfully. It can be failed with an exception.
https://github.com/line/armeria/blob/1869b40b2b0b2ce0af1bad76fcdc6e1d9742e3f5/core/src/main/java/com/linecorp/armeria/server/Http1RequestDecoder.java#L292-L293

What do you think of overriding `close()`, `close(Throwable)`, `abort()` and `abort(Throwable)` to log `requestFullyReceived`.
Additionally, should we update `requestFullyReceived` if it is not set when a request ends?
https://github.com/line/armeria/blob/1869b40b2b0b2ce0af1bad76fcdc6e1d9742e3f5/core/src/main/java/com/linecorp/armeria/common/logging/DefaultRequestLog.java#L1014-L1015

Reply:
What if we consider null `requestFullyReceived as a request that is **not** successfully received?
Do you think it doesn't provide clear meaning to the user? 🤔 
@ikhoon

Reply:
BTW, replying to your comment:

> There are two problems if we rely on isEndOfStream() to log requestFullyReceived
> - The last data for HTTP/1 does not have endOfStream flag.
> - Not all requests finish successfully. It can be failed with an exception.

 Do you think `StreamingDecodedHttpRequest` also have same problem like `AggregatingDecodedHttpRequest`?

Reply:
> What if we consider null `requestFullyReceived as a request that is not successfully received?
Do you think it doesn't provide clear meaning to the user? 

I think it's not clear and users can use the cause to see if there's an exception while receiving a request.

> Do you think StreamingDecodedHttpRequest also have same problem like AggregatingDecodedHttpRequest?

Yes, they have the same issues. 😉